packages/tkg-clusterclass-vsphere/bundle/config/upstream/base.yaml

```diff
@@ -2100,3 +2103,30 @@ spec:
             {{- range .additionalFQDN }}
             - {{ . }}
             {{- end }}
+  - name: systemWideProxy
+    enabledIf: '{{ .proxy.systemWide }}'
+    definitions:
+    - selector:
+        apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+        kind: KubeadmControlPlaneTemplate
+        matchResources:
+          controlPlane: true
+      jsonPatches:
+      - op: add
+        path: /spec/template/spec/preKubeadmCommands/-
+        value: echo "export HTTP_PROXY={{- .proxy.httpProxy }}" >> /etc/profile
+      - op: add
+        path: /spec/template/spec/preKubeadmCommands/-
```

Comment:
Path should be `path: /spec/template/spec/kubeadmConfigSpec/preKubeadmCommands/-`